internal/loader/configuration_client_manager.go

```diff
@@ -492,11 +493,22 @@ func newClientAssertionCredential(ctx context.Context, serviceAccountName string
 		return nil, err
 	}
 
-	clientId := serviceAccountObj.ObjectMeta.Annotations[AnnotationClientID]
-	tenantId := serviceAccountObj.Annotations[AnnotationTenantID]
+	tenantId, ok := os.LookupEnv(strings.ToUpper(AzureTenantId))
+	if !ok {
+		return nil, fmt.Errorf("no tenant ID specified. Check pod configuration or set TenantID in the options")
+	}
+
+	if _, ok := serviceAccountObj.Annotations[AnnotationClientID]; !ok {
+		return nil, fmt.Errorf("service account %s/%s does not have annotation %s", serviceAccountNamespace, serviceAccountName, AnnotationClientID)
+	}
+
+	if _, ok := serviceAccountObj.Annotations[AnnotationTenantID]; ok {
+		tenantId = serviceAccountObj.Annotations[AnnotationTenantID]
```

Comment:
What's the difference bewteen the `AzureTenantId` from pod config and `AnnotationTenantID` from service account?

Reply:
You can imagine that the `AzureTenantId` from pod is global, the `AnnotationTenantID` is for each specific service account. If users uses the same tenantID for all service account, it just need to use the global one, no need to specify in every service account

Reply:
to be honest, I cannot think of a case when a different tenant id is used. Technically it has to be the same tenantId.

Reply:
I do a bit change to this, hope this make sense to you